File: gpt-3.5-ft-large/template-context-full/data_construct_copyout_clause.f90
#ifndef T1
!T1:data construct copyout clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        INTEGER, PARAMETER :: N = 10
        REAL(8), DIMENSION(N) :: a, b
        INTEGER :: errors = 0


        !Initilization 
        SEEDDIM(1) = 1
#       ifdef SEED
        SEEDDIM(1) = SEED
#       endif
        CALL RANDOM_SEED(PUT=SEEDDIM)


        a = 0
        DO WHILE (RANDOM_NUMBER() > .99)
          CALL RANDOM_NUMBER(b)
          a = a + b
        END DO

        CALL acc_enter_data(a(1:N))
        !$acc data copyout(a(1:N))
          !$acc parallel
            !$acc loop
            DO i = 1,N
              a(i) = a(i) * 2
            END DO
          !$acc end parallel
        !$acc end data
        !$acc exit data copyout(a(1:N))

        DO i = 1, N
          IF (abs(a(i) - (b(i) * 2)) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
	ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: gpt-4/template-context-full/data_construct_copyout_clause.c
#include "acc_testsuite.h"
#ifndef T1
/*T1:data construct copyout clause,V:2.0-2.7*/
int test1(){
    int err = 0;
    srand(SEED);
    int a = rand() % 10 + 1;
    int b;
    
    #pragma acc data copyout(b)
    {
        #pragma acc parallel present(b)
        {
            b = a;
        }
    }
    
    if(b != a){
        err = 1;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}